### PR TITLE
Mobile-External-Auth —> Mobile

### PR DESCRIPTION
**Pull request by GGLozanov**
Implementation of external authentication on mobile, featuring Google and Facebook. GitHub is to be added in a later iteration. 

Several notes:
* This feature has also covered some general bug fixes and optimizations for the app
* There are minor bugs left to be addressed and they have their relevant issues opened (which will be fixed in the coming iterations)
* Also added special colour and text for cards displayed in match page whose users have sent a request to the current user.
* There was an attempt at Twitter sign-in but the Twitter developer API was bad so we decided to omit the third external auth option for now.

Requested review.

### CODE REVIEW

**Review by msimov02 — APPROVED** [review]
LGTM!

**Review by marianbitsov23 — APPROVED** [review]
Looks good to me!

File: Mobile/tues_pairs/lib/services/auth.dart [hunk 1]
```diff
@@ -17,16 +24,41 @@ class Auth {
...
+    return _auth.onAuthStateChanged.map((FirebaseUser user) {
+      // determine whether a user has signed in from google or facebook or any other platform
+      if(user != null && user.providerData != null) {
+        bool isExternPlatform = false;
+
+        for(var userInfo in user.providerData) {
+          switch(userInfo.providerId) {
+            case 'google.com':
+            case 'facebook.com':
+            case 'twitter.com':
```
Comment (marianbitsov23) [inline Mobile/tues_pairs/lib/services/auth.dart]: само това тук го промени на github

**Comment by GGLozanov** [comment]
Approved.

**Pull request merged**